### PR TITLE
[WIP] Use netcoreapp1.0 and publish with FSharp.Core.Unittests

### PR DESCRIPTION
**Pull request by KevinRansom**
Title kind of says it all

### CODE REVIEW

File: src/FSharpSource.targets [hunk 1]
```diff
@@ -548,10 +548,18 @@
     </ItemGroup>
   </Target>
 
-  <Target Name="dotnetrestore" BeforeTargets="Build" Condition=" '$(TargetFramework)' == 'coreclr' ">
+  <Target Name="nugetrestore" BeforeTargets="Build" Condition=" '$(TargetFramework)' == 'coreclr'  and '$(DOTNET_PUBLISH_PATH)' != 'true' ">
```
Comment (smoothdeveloper) [inline src/FSharpSource.targets]: is it `'$(DOTNET_PUBLISH)' != 'true'` or `'$(DOTNET_PUBLISH_PATH)' != 'true'`?

Comment (KevinRansom) [inline src/FSharpSource.targets]: @smoothdeveloper good eyes

File: src/fsharp/FSharp.Core.Unittests/project.json [hunk 2]
```diff
@@ -1,16 +1,11 @@
...
-    "NETStandard.Library": "1.6.0",
-    "System.Linq.Expressions": "4.1.0",
-    "System.Linq.Queryable": "4.0.1",
-    "System.Net.Requests": "4.0.11",
-    "System.Reflection.Emit": "4.0.1",
-    "System.Runtime.Loader": "4.0.0",
-    "System.Text.RegularExpressions": "4.1.0",
-    "System.Threading.Tasks.Parallel": "4.0.1",
-    "System.Threading.Thread": "4.0.0",
-    "System.Threading.ThreadPool": "4.0.10",
```
Comment (dsyme) [inline src/fsharp/FSharp.Core.Unittests/project.json]: Why do all these dependencies disappear?

Comment (KevinRansom) [inline src/fsharp/FSharp.Core.Unittests/project.json]: Yes.

**Pull request merged**